### PR TITLE
treewide: drop python2 support

### LINKED ISSUE #432: stop supporting python2
just realized that we even support python2! what the reason of behind this or we just haven't got a chance of updating https://github.com/scylladb/scylla-ccm/blob/e4c0589450f0df667cb2fd1e0f3c51afed2e54e5/setup.py#L32-L34 and remove all the code using `six`?

@fruch hi Israel, what do you think? i'd volunteer for this cleanup piecemeal with your blessings. as i miss the python3 features, and don't really want to use `six` or `__future__` in 2023.

### PR DESCRIPTION
**Pull request by tchaikov**
- setup, tox.ini: do not claim the python2.x support
- treewide: run 2to3 to the tree
- treewide: use f-string when appropriate
- treewide: do not use six module

Fixes #432

### CODE REVIEW

**Comment by tchaikov** [comment]
@fruch hi Israel, mind taking a look? in the meanwhile, i am running dtest (5% completed), so far so good.

File: README.md [hunk 1]
```diff
@@ -18,7 +18,6 @@ Requirements
 
 - A working python installation (tested to work with python 2.7).
```
Comment (fruch) [inline README.md:19]: this mention python 2.7, could be update to 3.11

File: ccmlib/cli_session.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from threading import Thread
 
 try:
```
Comment (fruch) [inline ccmlib/cli_session.py:4]: the try/except clause can be removed

File: tox.ini [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
```
Comment (fruch) [inline tox.ini:4]: in general I'm not sure we even used tox with ccm, and I don't know if this tox.ini is working
I think we can remove it

**Comment by fruch** [comment]
@tchaikov I've remember on the last place still using ccm with python 2
https://github.com/scylladb/python-driver/blob/master/.github/workflows/integration-tests-python2.yml

it's pending on the replacement of cqlsh in https://github.com/scylladb/scylladb/pull/11937

and then we can move forward with ditching python2 completely

File: tox.ini (hunk 3 above)
Comment (tchaikov) [reply tox.ini:4]: yup. also, nose is practically dead. but i'd prefer removing this in a separate PR. as this change is orthogonal to a more general cleanup which should drop unused testing harness. what do you think?

File: README.md (hunk 1 above)
Comment (tchaikov) [reply README.md]: sure, updated.

File: ccmlib/cli_session.py (hunk 2 above)
Comment (tchaikov) [reply ccmlib/cli_session.py]: dropped.

**Comment by tchaikov** [comment]
changelog

- dropped the no-more-needed try-catch
- update README.md to mention python 3.11 not python2.7

**Comment by tchaikov** [comment]
> @tchaikov I've remember on the last place still using ccm with python 2 https://github.com/scylladb/python-driver/blob/master/.github/workflows/integration-tests-python2.yml
> 
> it's pending on the replacement of cqlsh in [scylladb/scylladb#11937](https://github.com/scylladb/scylladb/pull/11937)
> 
> and then we can move forward with ditching python2 completely

@fruch just wanted to make sure i understand 100%: so we just need to wait for the promotion and merge of https://github.com/scylladb/scylladb/pull/11937 , right?

**Comment by fruch** [comment]
> > @tchaikov I've remember on the last place still using ccm with python 2 https://github.com/scylladb/python-driver/blob/master/.github/workflows/integration-tests-python2.yml
> > it's pending on the replacement of cqlsh in [scylladb/scylladb#11937](https://github.com/scylladb/scylladb/pull/11937)
> > and then we can move forward with ditching python2 completely
> 
> @fruch just wanted to make sure i understand 100%: so we just need to wait for the promotion and merge of [scylladb/scylladb#11937](https://github.com/scylladb/scylladb/pull/11937) , right?

Yes, so we won't be in a situation we can't create new driver releases (if fixes would be needed), once master stops using python2, we can continue with this remove the python2 CI from the driver (or maybe leaving it pinned down to old version of CCM, if we need fixes for 5.2/2023.1)

**Comment by tchaikov** [comment]
i see. thanks!

**Comment by tchaikov** [comment]
rebased against next HEAD.

**Comment by tchaikov** [comment]
@fruch hi Israel, now that https://github.com/scylladb/scylladb/pull/11937 has been merged. could you take another look at this PR?

File: tox.ini (hunk 3 above)
Comment (fruch) [reply tox.ini:4]: o.k. no rush

**Comment by tchaikov** [comment]
thanks. please let me know if anything i can help with.

**Comment by tchaikov** [comment]
@fruch hi Israel, to avoid the overhead of rebasing, if you don't have enough bandwidth, i can help on dropping the python2 support in https://github.com/scylladb/python-driver .

**Comment by fruch** [comment]
> @fruch hi Israel, to avoid the overhead of rebasing, if you don't have enough bandwidth, i can help on dropping the python2 support in https://github.com/scylladb/python-driver .  

Go for it, just sync with @avelanarius, to make sure it's agreed to move forward:
https://github.com/scylladb/python-driver/issues/214

**Comment by tchaikov** [comment]
@avelanarius Hi Piotr, could you please help conform this?

**Comment by avelanarius** [comment]
Reading the context of the thread, is the only actionable thing (for now) to disable Python2 CI in Python Driver? @Lorak-mmk will work on that.

**Comment by avelanarius** [comment]
Removing Python2 support from Python Driver PR: https://github.com/scylladb/python-driver/pull/215

**Comment by tchaikov** [comment]
> Reading the context of the thread, is the only actionable thing (for now) to disable Python2 CI in Python Driver? @Lorak-mmk will work on that.

yeah, at least this is what i learned from @fruch.

**Comment by tchaikov** [comment]
> Removing Python2 support from Python Driver PR: [scylladb/python-driver#215](https://github.com/scylladb/python-driver/pull/215)

thanks!

**Comment by tchaikov** [comment]
@fruch hi Israel, now that https://github.com/scylladb/python-driver/pull/215 has been merge, could you take a final look at this change and merge it if it still sane?

**Review by fruch — APPROVED** [review]
LGTM

File: tox.ini (hunk 3 above)
Comment (tchaikov) [reply tox.ini:4]: #446

**Pull request merged**